metrics.nim

```diff
@@ -166,6 +166,12 @@ when defined(metrics):
       result = result !& label.hash
     result = !$result
 
+  # Hash-based data types like OrderedSet don't just compare hashes to determine
+  # if a key is present, but also compare the keys themselves, so we need to
+  # override the `==` operator.
+  method `==`*(x, y: Collector): bool {.base.} =
+    x.hash == y.hash
```

Comment:
you need to do a full comparison here of the same fields that are used in `hash` - `hash` values may collide easily, specially on 32-bit

Reply:
Done.